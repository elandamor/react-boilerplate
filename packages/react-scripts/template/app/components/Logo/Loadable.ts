/**
 *
 * Asynchronously loads the component for Logo
 *
 */

import React from 'react';

export default React.lazy(() => import('./index'));
