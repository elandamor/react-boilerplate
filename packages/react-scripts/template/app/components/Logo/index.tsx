import React, { SFC } from 'react';
import classNames from 'classnames';
// Contexts
import { NetworkStatusContext } from '../../contexts/networkStatus.context';
// Styles
import Wrapper from './styles';

export interface IProps {
  className?: string;
  offline?: boolean;
};

/**
 * @render react
 * @name Logo component
 * @description Logo component.
 * @example
 * <Logo />
 */

const Logo: SFC<IProps> = ({ className }) => (
  <NetworkStatusContext.Consumer>
    {({ networkStatus }) => (
      <Wrapper
        className={classNames('', className)}
        offline={!networkStatus.isConnected}
      >
        <div id="react">
          <div id="react-inner"></div>
          <div id="react-innerdot"></div>
        </div>
      </Wrapper>
    )}
  </NetworkStatusContext.Consumer>

);

export default Logo;
