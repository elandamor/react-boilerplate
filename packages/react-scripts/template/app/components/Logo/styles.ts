import styled , { keyframes } from 'styled-components';
import { IProps } from './index';

const spin = keyframes`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(360deg) }
`;

const Wrapper = styled.div<IProps>`
  align-items: center;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  overflow: hidden;

  #react-inner {
    position: absolute;
      top: 0px;
      left: 0px;
      right: 0px;
      bottom: 0px;
      border-radius: 50%;
      border: 6px solid ${({ offline, theme }) => offline ? theme.palette.cardBorderColor : '#61dafb' };

      transform: rotate(120deg);
  }

  #react-innerdot {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    background: ${({ offline, theme }) => offline ? theme.palette.cardBorderColor : '#61dafb' };
    margin-top: -10px;
    margin-left: -10px;
  }

  #react {
    display: block;
    position: relative;
    width: 100px;
    height: 40px;
    border-radius: 50%;
    animation: ${spin} 10s linear infinite;
    z-index: 11;

    &:before {
      content: "";
      position: absolute;
      top: 0px;
      left: 0px;
      right: 0px;
      bottom: 0px;
      border-radius: 50%;
      border: 6px solid ${({ offline, theme }) => offline ? theme.palette.cardBorderColor : '#61dafb' };
      transform: rotate(-120deg);

    }

    &:after {
      content: "";
      position: absolute;
      top: 0px;
      left: 0px;
      right: 0px;
      bottom: 0px;
      border-radius: 50%;
      border: 6px solid ${({ offline, theme }) => offline ? theme.palette.cardBorderColor : '#61dafb' };
    }
  }
`;

export default Wrapper;
