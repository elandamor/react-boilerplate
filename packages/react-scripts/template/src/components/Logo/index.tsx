import React, { FC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

export interface IProps {
  className?: string;
}

/**
 * @render react
 * @name Logo component
 * @description Logo component.
 * @example
 * <Logo />
 */

const Logo: FC<IProps> = ({ className }) => (
  <Wrapper className={classNames('c-logo', className)}>
    <div id="react">
      <div id="react-inner" />
      <div id="react-innerdot" />
    </div>
  </Wrapper>
);

export default Logo;
