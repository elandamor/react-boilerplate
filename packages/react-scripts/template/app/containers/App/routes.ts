// tslint:disable:object-literal-sort-keys
import { Home } from '../../pages';

import { IRouteProps } from '../../components/Routes';

const routes: IRouteProps[] = [
  {
    exact: true,
    path: '/',
    component: Home,
  },
];

export default routes;
