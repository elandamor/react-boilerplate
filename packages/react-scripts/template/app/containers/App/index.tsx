import classNames from 'classnames';
import React, { Component, Suspense } from 'react';
import Measure from 'react-measure';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
// Components
import { ErrorBoundary, LoadingBar, Routes } from '../../components';
// Contexts
import NetworkStatusProvider from '../../contexts/networkStatus.context';
// Routes
import routes from '../../routes';
// Styles
import GlobalStyles from '../../global-styles';
import Wrapper from './styles';

import { breakpoints, makeDebugger } from '../../utils';
const debug = makeDebugger('App');

const defaultTheme = {
  breakpoints: {
    small: '600px',
    medium: '1024px',
    large: '1440px',
    xLarge: '1920px',
  },
};

/* tslint:disable:object-literal-sort-keys */
const themeLight = {
  ...defaultTheme,
  isDark: false,
  palette: {
    bodyBackground: '#FAFAFA',
    brandPrimary: '#FFFFFF',
    cardBackground: '#FFFFFF',
    cardBorderColor: '#E4E6E9',
  },
};
/* tslint:enable:object-literal-sort-keys */

export interface IProps extends RouteComponentProps<any> {}

interface IState {
  bounds: {
    height: number;
    width: number;
  };
  theme: object;
}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */
class App extends Component<IProps, IState> {
  protected componentIsMounted: boolean;
  protected previousLocation: object;

  constructor(props: IProps) {
    super(props);

    this.state = {
      bounds: {
        height: 0,
        width: 0,
      },
      theme: themeLight,
    };

    this.previousLocation = props.location;
  }

  public componentDidMount() {
    this.componentIsMounted = true;
    debug('componentIsMounted');
  }

  public componentWillUpdate(nextProps: IProps) {
    const { location } = this.props;

    if (
      nextProps.history.action !== 'POP' &&
      (!location.state || !location.state.modal)
    ) {
      this.previousLocation = this.props.location;
    }
  }

  public componentWillUnmount() {
    this.componentIsMounted = false;
  }

  public setState(nextState: any, cb?: () => void) {
    if (this.componentIsMounted) {
      super.setState(nextState, cb);
    }
  }

  public render() {
    const { location } = this.props;
    const {
      bounds: { width },
    } = this.state;

    const isModal = !!(
      location.state &&
      location.state.modal &&
      this.previousLocation !== location
    );

    return (
      <ThemeProvider theme={this.state.theme}>
        <NetworkStatusProvider>
          <Measure
            bounds
            onResize={(contentRect) => {
              this.setState({ bounds: contentRect.bounds });
            }}
          >
            {({ measureRef }) => (
              <Wrapper
                className={classNames('c-app__container', breakpoints(width))}
                // @ts-ignore
                ref={measureRef}
              >
                <GlobalStyles />
                <ErrorBoundary>
                  <Suspense fallback={<LoadingBar loading />}>
                    <Routes
                      location={isModal ? this.previousLocation : location}
                      routes={routes}
                    />
                  </Suspense>
                </ErrorBoundary>
              </Wrapper>
            )}
          </Measure>
        </NetworkStatusProvider>
      </ThemeProvider>
    );
  }
}

export default withRouter(App);
