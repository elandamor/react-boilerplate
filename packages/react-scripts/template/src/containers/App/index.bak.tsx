import React, { Suspense } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import get from 'lodash/get';
import merge from 'lodash/merge';
import { useTransition } from 'react-spring';
// Components
import { ErrorBoundary, Header, LoadingBar, Routes, AnimatedWrapper } from '@app/components';
// Contexts
import NetworkStatusProvider from '@app/contexts/networkStatus.context';
// Routes
import routes from '@app/routes';
// Styles
import { Wrapper } from './styles';

import GlobalStyles from '@app/global-styles';

import baseTheme from '@app/theme';
import { useDarkMode, useRouter } from '@app/hooks';

// import { makeDebugger } from '@app/utils';
// const debug = makeDebugger('App');

export interface IAppProps extends RouteComponentProps {}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

const App = () => {
  const { location } = useRouter();
  const [darkMode] = useDarkMode();

  // Merge the color mode with the base theme to create a new theme object
  const getTheme = (mode: string) => merge({}, baseTheme, {
    colors: get(baseTheme.colors.modes, mode, baseTheme.colors),
    isDark: true,
  })

  const theme = darkMode ? getTheme('dark') : baseTheme;

  const pageTransitions = useTransition(location, location => location.pathname, {
    from: { transform: 'translateY(100px)', opacity: 0 },
    enter: { transform: 'translateY(0)', opacity: 1 },
    leave: { transform: 'translateY(100px)', opacity: 0 },
  })

  return (
    <ThemeProvider theme={theme}>
      <NetworkStatusProvider>
        <Wrapper>
          <Normalize />
          <GlobalStyles />
          <ErrorBoundary>
            <Header alignItems="center" flex="none" />
          </ErrorBoundary>
          {
            pageTransitions.map(({ item, props, key }) => (
              <AnimatedWrapper key={key} style={props}>
                <ErrorBoundary>
                  <Suspense fallback={<LoadingBar />}>
                    <Routes routes={routes} location={item} />
                  </Suspense>
                </ErrorBoundary>
              </AnimatedWrapper>
            ))
          }
        </Wrapper>
      </NetworkStatusProvider>
    </ThemeProvider>
  );
}

export default withRouter(App);
