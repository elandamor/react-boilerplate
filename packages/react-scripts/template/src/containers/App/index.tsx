import React, { Suspense } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import get from 'lodash/get';
import merge from 'lodash/merge';
// Components
import { ErrorBoundary, Header, LoadingBar, Routes } from '@app/components';
// Contexts
import NetworkStatusProvider from '@app/contexts/networkStatus.context';
// Routes
import routes from '@app/routes';
// Styles
import { Wrapper } from './styles';

import GlobalStyles from '@app/global-styles';

import baseTheme from '@app/theme';
import { useDarkMode } from '@app/hooks';

// import { makeDebugger } from '@app/utils';
// const debug = makeDebugger('App');

export interface IAppProps extends RouteComponentProps {}

export const AppContext = React.createContext({});

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

const App = () => {
  const [darkMode, setDarkMode] = useDarkMode();

  // Merge the color mode with the base theme to create a new theme object
  const getTheme = (mode: string) => merge({}, baseTheme, {
    colors: get(baseTheme.colors.modes, mode, baseTheme.colors),
    isDark: true,
  })

  const theme = darkMode ? getTheme('dark') : baseTheme;

  return (
    <ThemeProvider theme={theme}>
      <NetworkStatusProvider>
        <AppContext.Provider value={{ darkMode, setDarkMode }}>
          <Wrapper>
            <Normalize />
            <GlobalStyles />
            <ErrorBoundary>
              <Header alignItems="center" flex="none" />
            </ErrorBoundary>
            <ErrorBoundary>
              <Suspense fallback={<LoadingBar />}>
                <Routes routes={routes} />
              </Suspense>
            </ErrorBoundary>
          </Wrapper>
        </AppContext.Provider>
      </NetworkStatusProvider>
    </ThemeProvider>
  );
}

export default withRouter(App);
