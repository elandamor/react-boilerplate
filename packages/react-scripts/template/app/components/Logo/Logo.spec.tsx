// Logo.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import 'whatwg-fetch';

import { ThemeProvider } from 'styled-components';

import Logo from './index';
import { themeLight } from '../../containers/App';

const theme = {
  ...themeLight,
};

describe('<Logo />', () => {
  it('should display a logo', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Logo />
      </ThemeProvider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
