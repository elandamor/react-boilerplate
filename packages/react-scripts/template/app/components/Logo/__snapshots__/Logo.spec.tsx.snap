// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should display a logo 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  overflow: hidden;
}

.c0 #react-inner {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  border-radius: 50%;
  border: 6px solid #61dafb;
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
}

.c0 #react-innerdot {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background: #61dafb;
  margin-top: -10px;
  margin-left: -10px;
}

.c0 #react {
  display: block;
  position: relative;
  width: 100px;
  height: 40px;
  border-radius: 50%;
  -webkit-animation: kOBPdp 10s linear infinite;
  animation: kOBPdp 10s linear infinite;
  z-index: 11;
}

.c0 #react:before {
  content: "";
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  border-radius: 50%;
  border: 6px solid #61dafb;
  -webkit-transform: rotate(-120deg);
  -ms-transform: rotate(-120deg);
  transform: rotate(-120deg);
}

.c0 #react:after {
  content: "";
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  border-radius: 50%;
  border: 6px solid #61dafb;
}

<div
  className="c0"
>
  <div
    id="react"
  >
    <div
      id="react-inner"
    />
    <div
      id="react-innerdot"
    />
  </div>
</div>
`;
